LibCore/CProcess.cpp

```diff
@@ -0,0 +1,104 @@
+#include "CProcess.h"
+
+#include <sched.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <signal.h>
+#include <unistd.h>
+
+void handle_sigchld(int)
+{
+    dbgprintf("CProcess(%d) Got SIGCHLD\n", getpid());
+    int pid = waitpid(-1, nullptr, 0);
+    dbgprintf("CProcess(%d) waitpid() returned %d\n", getpid(), pid);
+    ASSERT(pid > 0);
+}
+
+CProcess::CProcess(int pid)
+    : m_pid(pid)
+{
+    static bool signal_handled = false;
+    if (!signal_handled) {
+        signal_handled = true;
+        signal(SIGCHLD, handle_sigchld);
+    }
+}
+
+CProcess::~CProcess()
+{
+    dbgprintf("Destroying %d\n", m_pid);
+    kill();
+}
+
+void CProcess::set_priority(Priority priority)
+{
+    int prio = -1;
+    switch (priority) {
+    case Priority::Highest:
+        prio = sched_get_priority_max(SCHED_OTHER);
+        break;
+    case Priority::Normal:
+        return;
+        break;
+    case Priority::Lowest:
+        prio = sched_get_priority_min(SCHED_OTHER);
+        break;
+    }
+
+    if (priority != Priority::Normal) {
+        dbgprintf("Setting PID %d to prio %d\n", m_pid, prio);
+        struct sched_param p;
+        p.sched_priority = prio;
+        int ret = sched_setparam(m_pid, &p);
+        ASSERT(ret == 0);
+    }
+}
+
+CProcess* CProcess::start_detached(const StringView& program, const Vector<StringView>& arguments, Priority priority)
+{
+    pid_t pid = 0;
+
+    while (true) {
+        dbgprintf("Forking for %s...\n", program.characters());
```

Comment:
You cannot do that, `StringView::characters()` is not a null-terminated C string

Reply:
I'm not sure I understand your point.

The convention established elsewhere is that we take StringView as arguments, and store it persistently if needed in String - and thus far that has worked just fine.

Yes, if you pass garbage as a StringView, you're going to have a bad time. So? The same goes for pretty much everywhere else in Serenity where StringView is used too.

Reply:
No, sure, you can take `StringView`, but you cannot assume `StringView`'s `characters()` will return a null-terminated string, and even if it does, you cannot assume this C string will equal the actual contents of the string view.

Imagine a shell is doing this:
```c++
String path = getenv("PATH");
for (const StringView& dir : path.split_view(':')) {
    something(dir);
}
```
this is a perfectly valid way to create `StringView`s — not garbage values — but they won't be null-terminated, they will point into the middle of the original `String path` buffer.

So if `something()` assumes they are, it won't work correctly. Here are [two bugs](https://github.com/SerenityOS/serenity/commit/e585ed48b0b3e89a8bb4a805b568b7627c42c72b) [caused by this](https://github.com/SerenityOS/serenity/commit/75df45d709445c8a74ea8b7be5c60ef3fe852a90) I've fixed already, let's not add new ones :smile: